api/main.py

```diff
@@ -27,8 +28,18 @@
 
 @metrics.log_metrics(capture_cold_start_metric=True)
 def handler(event, context):
+    global clamd_pid
+
+    if environ.get("CI") is None and not is_clamd_running():
+        if clamd_pid is not None:
+            kill_process_by_pid(clamd_pid)
```

Comment:
Something to watch for here is that I've seen cases where we can end up with unkillable zombie processes, depending on how they're started by the lambda handler.

It's not a huge problem, but it could lead to pid exhaustion in edge cases if the lambda was never able to shutdown and for some reason we ended up with a lot of zombie daemons.